java/sage/FSManager.java

```diff
@@ -478,7 +492,12 @@ public void releaseISOMount(java.io.File mountDir)
         }
       }
       else
-        IOUtils.exec2(new String[] { "umount", mountDir.getAbsolutePath() }, false);
+      {
+        if (Sage.LINUX_IS_ROOT)
+          IOUtils.exec2(new String[]{"umount", mountDir.getAbsolutePath()}, false);
+        else
+          IOUtils.exec2(new String[]{"sudo","umount", mountDir.getAbsolutePath()}, false);
```

Comment:
To clarify, it will timeout eventually, but it could take about 5 minutes or more depending on the system defaults. In the mean time a few threads are tied up for each mount attempt.